java/src/main/java/ai/rapids/cudf/ParquetColumnWriterOptions.java

```diff
@@ -0,0 +1,406 @@
+      }
+      return (T) this;
+    }
+
+    /**
+     * Set a Decimal child meta data
+     * @param name
+     * @param precision
+     * @return this for chaining.
+     */
+    public T withDecimalColumn(String name, int precision, boolean nullable) {
+      children.add(withDecimal(name, precision, nullable));
+      return (T) this;
+    }
+
+    /**
+     * Set a Decimal child meta data
+     * @param name
+     * @param precision
+     * @return this for chaining.
+     */
+    public T withNullableDecimalColumn(String name, int precision) {
+      withDecimalColumn(name, precision, true);
+      return (T) this;
+    }
+
+    /**
+     * Set a Decimal child meta data
+     * @param name
+     * @param precision
+     * @return this for chaining.
+     */
+    public T withDecimalColumn(String name, int precision) {
+      withDecimalColumn(name, precision, false);
+      return (T) this;
+    }
+
+    /**
+     * Set a timestamp child meta data
+     * @param name
+     * @param isInt96
+     * @return this for chaining.
+     */
+    public T withTimestampColumn(String name, boolean isInt96, boolean nullable) {
+      children.add(withTimestamp(name, isInt96, nullable));
+      return (T) this;
+    }
+
+    /**
+     * Set a timestamp child meta data
+     * @param name
+     * @param isInt96
+     * @return this for chaining.
+     */
+    public T withTimestampColumn(String name, boolean isInt96) {
+      withTimestampColumn(name, isInt96, false);
+      return (T) this;
+    }
+
+    /**
+     * Set a timestamp child meta data
+     * @param name
+     * @param isInt96
+     * @return this for chaining.
+     */
+    public T withNullableTimestampColumn(String name, boolean isInt96) {
+      withTimestampColumn(name, isInt96, true);
+      return (T) this;
+    }
+  }
+
+  public static class StructBuilder extends AbstractStructBuilder<StructBuilder> {
+    public StructBuilder(String name, boolean isNullable) {
+      super(name, isNullable);
+    }
+
+    public ParquetStructColumnWriterOptions build() {
+      return new ParquetStructColumnWriterOptions(this);
+    }
+  }
+
+  public abstract static class AbstractStructBuilder<T extends AbstractStructBuilder> extends NestedBuilder<T> {
+    /**
+     * Builder specific to build a Struct meta
+     * @param name
+     */
+    public AbstractStructBuilder(String name, boolean isNullable) {
+      super(name, isNullable);
+    }
+
+    protected AbstractStructBuilder() {
+      super();
+    }
+  }
+
+  public static class ListBuilder extends NestedBuilder<ListBuilder> {
+
+    public ListBuilder(String name, boolean isNullable) {
+      super(name, isNullable);
+    }
+
+    public ParquetListColumnWriterOptions build() {
+      return new ParquetListColumnWriterOptions(this);
+    }
+  }
+
+  public static ListBuilder listBuilder(String name) {
+    return new ListBuilder(name, true);
+  }
+
+  public static ListBuilder listBuilder(String name, boolean isNullable) {
```

Comment:
we need java docs here. They really need to describe how a list builder is different from a StructBuilder and that the name of the child is ignored.

Reply:
I Still don't see it documented anywhere how a list builder works or that the name for it's children is ignored.